packages/EasyCodingStandard/packages/ChangedFilesDetector/src/Cache/Simple/FilesystemCacheFactory.php

```diff
@@ -12,13 +11,19 @@ final class FilesystemCacheFactory
      */
     private $cacheDirectory;
 
-    public function __construct(string $cacheDirectory)
+    /**
+     * @var string
+     */
+    private $cacheNamespace;
+
+    public function __construct(string $cacheDirectory, string $cacheNamespace)
     {
         $this->cacheDirectory = $cacheDirectory;
+        $this->cacheNamespace = $cacheNamespace;
```

Comment:
@gnutix I'm thinking of a solution. This doesn't have to a parameter, but a service.

```php
class CacheNamespaceProvider
{
    public function provide(): string
    {
        // 1. current getcwd-based namespace

        // 2. fallback to Docker-related unique path namespace
        // you need to figure this out, what is unique 
    }
}
```

Reply:
> you need to figure this out, what is unique 

I think the whole point of the issue is this : what fallback to use. However, I don't understand your comment : 

> fallback to Docker-related unique path namespace

I'm not using Docker, but ecs.phar directly when I encounter the issue of getcwd() returning false. Which makes me wonder : can you reproduce on your machine ? (running ecs.phar)

Reply:
Ah, I though it's Docker related.
In that case ecs.phar related.

I don't recall such issue.
Test is needed in https://github.com/symplify/easy-coding-standard-prefixed

Reply:
Before having to figure out how to write a test that breaks getcwd, could you simply download the phar and run it on your computer, just to know if you too have the issue (and thus probably everyone) or not ? (If you have a php 7.2 setup on your laptop, might be worth checking on this version too - though I doubt that php version would change the behavior of getcwd)

Reply:
Sorry, I don't have time for testing. I working on something else.

Reply:
Also, test is needed not just for this moment, but also to prevent future regresions.

Reply:
Sure, a test is better for all reasons. But my question here is quite simple : is `ecs.phar` broken for everyone and it's just not covered by any tests, or is it just me ? Or to get that answer, the simplest way is by far that **anyone** just tries it locally on their computer and report here... Takes literally 10 seconds. Not much of a request I guess...

Reply:
I tried again with the latest version, and I might have said something terribly wrong. **It is Docker related**. Running `php ecs.phar check -c easy-coding-standard.yaml` worked (on both PHP 7.2, 7.3 and 7.4) locally, while running it through Docker failed : 

```
docker run --rm -i -t -v "/home/gnutix/dev/myproject:/app" --net=myproject-760ee_default -e APP_ENV=dev -e APP_DEBUG=0 -e USER_ID=1000 -e GROUP_ID=1000 -v "/home/gnutix/dev/myproject/scripts/docker/php-cli/php.ini:/usr/local/etc/php/php.ini" eu.gcr.io/myproject/php:7.2.28-a php ecs.phar check  --ansi -c easy-coding-standard.yml
```